client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -147,25 +149,33 @@ const PatternAssembler: Step = ( { navigation } ) => {
 		] );
 	};
 
-	const onSelect = ( pattern: Pattern | null ) => {
-		if ( pattern ) {
-			if ( 'header' === showPatternSelectorType ) {
-				setHeader( pattern );
-			}
-			if ( 'footer' === showPatternSelectorType ) {
-				setFooter( pattern );
-			}
-			if ( 'section' === showPatternSelectorType ) {
-				addSection( pattern );
+	const onSelect = ( selectedPattern: Pattern ) => {
+		if ( 'header' === showPatternSelectorType ) {
+			setHeader( selectedPattern );
+		}
+		if ( 'footer' === showPatternSelectorType ) {
+			setFooter( selectedPattern );
+		}
+		if ( 'section' === showPatternSelectorType ) {
+			if ( sectionPosition !== null ) {
+				replaceSection( selectedPattern );
+			} else {
+				addSection( selectedPattern );
```

Comment:
[nit] We could call the track event earlier and return here immediately. What do you think?